src/main/scala/com/github/gvolpe/scalar2019/tagless/alt/laws.scala

```diff
@@ -0,0 +1,34 @@
+package com.github.gvolpe.scalar2019.tagless.alt
+
+import laws._
+
+object laws {
+  final case class IsEq[A](lhs: A, rhs: A)
+
+  implicit final class IsEqArrow[A](private val lhs: A) extends AnyVal {
+    def <->(rhs: A): IsEq[A] = IsEq(lhs, rhs)
+  }
+}
+
+trait MkDepLaws[F[- _, _], R] {
+  def M: MkDep[F, R]
+
+  /*
+   * Feeding R to F[R, A] eliminates R ang gives you F[Any, A] which is just F[A]
+   */
+  def elimination[A](fra: F[R, A], env: R, fa: F[Any, A]) = M[A](fra)(env) <-> fa
+
+}
+
+trait DependencyLaws {
+
+  def identity[F[_], A](fa: F[A])(implicit ev: Dependency[F, F]) =
```

Comment:
Basically this is `CategoryK` for functors.

Reply:
In `cats` is this `FunctionK.id`?

Reply:
Yes. Although presumably, here we are requiring `F` and `G` be functors.